pageserver/src/virtual_file/owned_buffers_io/write.rs

```diff
@@ -219,119 +237,105 @@ impl Buffer for BytesMut {
     }
 }
 
-impl OwnedAsyncWriter for Vec<u8> {
-    async fn write_all<Buf: IoBuf + Send>(
-        &mut self,
-        buf: FullSlice<Buf>,
-        _: &RequestContext,
-    ) -> std::io::Result<(usize, FullSlice<Buf>)> {
-        self.extend_from_slice(&buf[..]);
-        Ok((buf.len(), buf))
+impl Buffer for IoBufferMut {
+    type IoBuf = IoBuffer;
+
+    fn cap(&self) -> usize {
+        self.capacity()
+    }
+
+    fn extend_from_slice(&mut self, other: &[u8]) {
+        IoBufferMut::extend_from_slice(self, other);
+    }
+
```

Comment:
The trait definition of `Buffer::extend_from_slice` says this function panics if there's not enough capacity left to accomodate `other`.
This implementation here reserves more capacity instead.
Not sure if it matters.

But so far, the only other user is a `From` impl where the condition holds true.

So, might as well panic instead of reserve?

Reply:
bf9a6d0f4c2ae95610322056724a8a181da3d521: I made Buffer::extend_from_slice panic, but retains the reserve behavior for `IoBufferMut::extend_from_slice`, similar to `Vec::extend_from_slice` and `BytesMut::extend_from_slice`.